datadog/resource_datadog_logs_custom_pipeline.go

```diff
@@ -134,6 +138,113 @@ var arithmeticProcessor = &schema.Schema{
 	},
 }
 
+var arrayProcessor = &schema.Schema{
+	Type:        schema.TypeList,
+	MaxItems:    1,
+	Description: "Array Processor. More information can be found in the [official docs](https://docs.datadoghq.com/logs/processing/processors/?tab=ui#array-processor)",
+	Optional:    true,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "Your pipeline name.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"is_enabled": {
+				Description: "Boolean value to enable your pipeline.",
+				Type:        schema.TypeBool,
+				Optional:    true,
+			},
+			"operation": {
+				Description: "Operation to perform on the array.",
+				Type:        schema.TypeList,
+				Required:    true,
+				MaxItems:    1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"append": {
+							Description: "Operation that appends a value to a target array attribute.",
+							Type:        schema.TypeList,
+							Optional:    true,
+							MaxItems:    1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"source": {
+										Description: "Attribute path containing the value to append.",
+										Type:        schema.TypeString,
+										Required:    true,
+									},
+									"target": {
+										Description: "Attribute path of the array to append to.",
+										Type:        schema.TypeString,
+										Required:    true,
+									},
+									"preserve_source": {
+										Description: "Remove or preserve the remapped source element.",
```

Comment:
The description doesn't say which one of `Remove` or `preserve` is done when the flag is `true` or `false` but hopefully the flag name makes it clear and it's the same description as the `preserve_source` of the Remapper, so 👍